File: client/src/main/java/me/ftvkyo/fuzzy_broccoli/client/graphics/primitives/VertexTextured.java
package me.ftvkyo.fuzzy_broccoli.client.graphics.primitives;


import org.joml.Vector2f;
import org.joml.Vector3f;

import java.nio.FloatBuffer;


/**
 * Extension of simple vertex that can store vertex texture binding.
 */
public class VertexTextured extends VertexSimple {

    private Vector2f texture;


    public VertexTextured(float x, float y, float z, float textureX, float textureY) {
        super(x, y, z);
        texture = new Vector2f(textureX, textureY);
    }


    public VertexTextured(Vector3f pos, Vector2f tex) {
        super(pos);
        texture = tex;
    }


    /**
     * Get number of components used to store texture info.
     *
     * @return number of texture components.
     */
    public int componentsPerTexture() {
        return 2;
    }


    public void get(FloatBuffer fb) {
        super.get(fb);
        texture.get(fb);
        fb.position(fb.position() + componentsPerTexture());
    }
}

File: client/src/main/java/me/ftvkyo/fuzzy_broccoli/client/graphics/primitives/VertexTextured.kt
package me.ftvkyo.fuzzy_broccoli.client.graphics.primitives


import org.joml.Vector2f
import org.joml.Vector3f

import java.nio.FloatBuffer


/**
 * Extension of simple vertex that can store vertex texture binding.
 */
class VertexTextured : VertexSimple {

    private val texture: Vector2f


    constructor(x: Float, y: Float, z: Float, textureX: Float, textureY: Float) : super(x, y, z) {
        texture = Vector2f(textureX, textureY)
    }


    constructor(pos: Vector3f, tex: Vector2f) : super(pos) {
        texture = tex
    }


    /**
     * Get number of components used to store texture info.
     *
     * @return number of texture components.
     */
    fun componentsPerTexture(): Int {
        return 2
    }


    override fun get(fb: FloatBuffer) {
        super.get(fb)
        texture.get(fb)
        fb.position(fb.position() + componentsPerTexture())
    }
}
